src/Paprika.Tests/Data/NibblePathTests.cs

```diff
@@ -1,14 +1,129 @@
 using System.Buffers.Binary;
 using FluentAssertions;
 using NUnit.Framework;
+using Paprika.Crypto;
 using Paprika.Data;
+using Paprika.Utils;
 
 // ReSharper disable HeapView.BoxingAllocation
 
 namespace Paprika.Tests.Data;
 
 public class NibblePathTests
 {
+    [Test]
+    public void Get_Singles()
+    {
+        // Expected values
+        byte[] expectedSingles =
+        [
+            0x00, 0x11, 0x22, 0x33, 0x44, 0x55, 0x66, 0x77,
+            0x88, 0x99, 0xAA, 0xBB, 0xCC, 0xDD, 0xEE, 0xFF
+        ];
+
+        // Get the actual values from the NibblePath.Singles property
+        var actualSingles = NibblePath.SinglesForTests;
+
+        // Check if the length is correct
+        actualSingles.Length.Should().Be(expectedSingles.Length);
+
+        // Check if each value is correct
+        for (var i = 0; i < expectedSingles.Length; i++)
+        {
+            actualSingles[i].Should().Be(expectedSingles[i]);
+        }
+    }
+
+    [Test]
+    public void Get_Single([Values(0, 1)] int odd)
+    {
+        for (byte nibble = 0; nibble <= NibblePath.NibbleMask; nibble++)
+        {
+            var nibblePath = NibblePath.Single(nibble, odd);
+
+            // Check the length
+            nibblePath.Length.Should().Be(1);
+
+            // Check the nibble value
+            nibblePath[0].Should().Be(nibble);
+
+            // Check the oddity
+            bool isOdd = nibblePath.Oddity == 1;
```

Comment:
There should be `IsOdd` property to assert more easily?

Reply:
This was to add `Oddity` for test coverage, again this and above are not actually required, and I remove these if you say so :)